### PR TITLE
Fix unit tests mocking

### PR DESCRIPTION
**Pull request by s0undt3ch**
### What does this PR do?
Fixes the unit tests mocking by not polluting the loader modules namespace and instead patch.

### CODE REVIEW

**Review by s0undt3ch — COMMENTED** [review]
Do not merge. I'm still about halfway but I want to see what Jenkins has to say about the changes so far.

**Comment by s0undt3ch** [comment]
Bah! Lint again! Wait!!! How did that test pass locally..... I'll fix that one in the evening...

**Comment by s0undt3ch** [comment]
Having fun yet @cachedout? :)

**Comment by cachedout** [comment]
Always!!

**Comment by s0undt3ch** [comment]
K rebased and fixed lint issue.

**Comment by s0undt3ch** [comment]
Bah, this needs a rebase.

**Comment by s0undt3ch** [comment]
Ok. I should be done now.

**Comment by s0undt3ch** [comment]
Go Go Jenkins!

**Comment by rallytime** [comment]
@s0undt3ch Just a couple more lint errors that snuck in here. :)

https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/9873/violations/file/tests/support/mixins.py/

**Comment by s0undt3ch** [comment]
Whoa! Those are not line errors, those are bugs! ;)

**Comment by s0undt3ch** [comment]
*lint

**Comment by s0undt3ch** [comment]
Should be fixed now.

**Comment by s0undt3ch** [comment]
Go Go Jenkins!

**Comment by s0undt3ch** [comment]
Go Go Jenkins!

**Comment by s0undt3ch** [comment]
Rebased to resolve conflicts

**Comment by s0undt3ch** [comment]
Rebased to resolve conflicts

**Review by terminalmage — APPROVED** [review]
This is great. :heart: :+1:

**Comment by s0undt3ch** [comment]
Rebased to resolve conflicts, again.

File: templates/module/salt/modules/{{module_name}}.py
Comment (s0undt3ch) [inline templates/module/salt/modules/{{module_name}}.py]: Fixed locally. Let me known if/when you're done with the review so I can force push the changes.

**Comment by s0undt3ch** [comment]
Rebased to resolve conflicts, yet again.

**Comment by s0undt3ch** [comment]
Go Go Jenkins!

File: templates/module/salt/modules/{{module_name}}.py
Comment (terminalmage) [inline templates/module/salt/modules/{{module_name}}.py]: Sorry, didn't see this comment. I approved the changes so we're good to go pending your latest rebase.

**Comment by s0undt3ch** [comment]
Go Go Jenkins!

**Comment by s0undt3ch** [comment]
🎉

**Pull request merged**
